{
  super.beginExceptionGroup();
  if (!(context().block instanceof RuleBlock)) {
    tool.panic("beginExceptionGroup called outside of rule block");
  }
}
